src/prepare.c

```diff
@@ -131,32 +135,151 @@ static void prepare_class_function_table(zend_class_entry *candidate, zend_class
-    
-	if (candidate->parent) {
-		if (zend_hash_index_exists(&PTHREADS_ZG(resolve), (zend_ulong) candidate->parent)) {
-			prepared->parent = zend_hash_index_find_ptr(&PTHREADS_ZG(resolve), (zend_ulong) candidate->parent);
-		} else prepared->parent = pthreads_prepared_entry_internal(thread, candidate->parent, prepare_static_members);
+static void prepare_class_property_table(pthreads_object_t* thread, zend_class_entry *candidate, zend_class_entry *prepared) {
+
+	zend_property_info *info;
+	zend_string *name;
+	ZEND_HASH_FOREACH_STR_KEY_PTR(&candidate->properties_info, name, info) {
+		zend_property_info dup = *info;
+		if (info->doc_comment) {
+			if (thread->options & PTHREADS_INHERIT_COMMENTS) {
+				dup.doc_comment = zend_string_new(info->doc_comment);
+			} else dup.doc_comment = NULL;
+		}
+
+		if (info->ce) {
+			if (info->ce == candidate) {
+				dup.ce = prepared;
+			} else dup.ce = pthreads_prepared_entry(thread, info->ce);
+		}
+
+		if (!zend_hash_str_add_mem(&prepared->properties_info, name->val, name->len, &dup, sizeof(zend_property_info))) {
+			if (dup.doc_comment)
+				zend_string_release(dup.doc_comment);
+		}
+	} ZEND_HASH_FOREACH_END();
+
+	if (candidate->default_properties_count) {
+		int i;
+
+		if(prepared->default_properties_table != NULL) {
+			efree(prepared->default_properties_table);
+		}
+		prepared->default_properties_table = emalloc(
+			sizeof(zval) * candidate->default_properties_count);
+
+		memcpy(
+			prepared->default_properties_table,
+			candidate->default_properties_table,
+			sizeof(zval) * candidate->default_properties_count);
+
+		for (i=0; i<candidate->default_properties_count; i++) {
+			if (Z_REFCOUNTED(prepared->default_properties_table[i])) {
+				pthreads_store_separate(
+					&candidate->default_properties_table[i],
+					&prepared->default_properties_table[i], 1);
+			}
+		}
+		prepared->default_properties_count = candidate->default_properties_count;
+	} else prepared->default_properties_count = 0;
+} /* }}} */
+
+/* {{{ */
+static void prepare_class_handlers(zend_class_entry *candidate, zend_class_entry *prepared) {
+	uint umethod = 0;
+	void *usources[6] = {
+		candidate->create_object,
+		candidate->serialize,
+		candidate->unserialize,
+		candidate->get_iterator,
+		candidate->interface_gets_implemented,
+		candidate->get_static_method
+	};
+
+	do {
+		if (usources[umethod]) {
+			switch(umethod){
+				case 0: prepared->create_object = candidate->create_object; break;
+				case 1: prepared->serialize = candidate->serialize; break;
+				case 2: prepared->unserialize = candidate->unserialize; break;
+				case 3: {
+					prepared->get_iterator = candidate->get_iterator;
+					prepared->iterator_funcs = candidate->iterator_funcs;
+				} break;
+				case 4: prepared->interface_gets_implemented = candidate->interface_gets_implemented; break;
+				case 5: prepared->get_static_method = candidate->get_static_method; break;
+			}
+		}
+	} while(++umethod < 6);
+} /* }}} */
```

Comment:
I realise this is just copied code, but the whole function body could be compacted to just 7 lines of code:
```
prepared->create_object = candidate->create_object;
prepared->serialize = candidate->serialize;
prepared->unserialize = candidate->unserialize;
prepared->get_iterator = candidate->get_iterator;
prepared->iterator_funcs = candidate->iterator_funcs;
prepared->interface_gets_implemented = candidate->interface_gets_implemented;
prepared->get_static_method = candidate->get_static_method;
```